src/test/java/utils/BaseTest.java

```diff
@@ -1,39 +1,71 @@
 package utils;
 
+import exceptions.TestExecutionException;
 import manager.PageFactoryManager;
 import org.openqa.selenium.WebDriver;
-import org.openqa.selenium.chrome.ChromeDriver;
-import org.testng.annotations.AfterClass;
-import org.testng.annotations.BeforeClass;
-import pages.AlertAndPopupPage;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Parameters;
+import pages.AccordionItemsPage;
+import pages.ActionsPage;
+import pages.LoginPortalPage;
+import pages.PopupAndAlertsPage;
+import pages.OptionsPage;
 import pages.HomePage;
 
-import static io.github.bonigarcia.wdm.WebDriverManager.chromedriver;
+import java.net.MalformedURLException;
+import java.net.URL;
 
 public class BaseTest {
-  static PageFactoryManager pageFactoryManager;
   protected static HomePage homePage;
-  protected static AlertAndPopupPage alertAndPopupPage;
+  protected static PopupAndAlertsPage popupAndAlertsPage;
+  protected static AccordionItemsPage accordionItemsPage;
+  protected static OptionsPage optionsPage;
+  protected static LoginPortalPage loginPortalPage;
+  protected static ActionsPage actionsPage;
+  protected static PageFactoryManager pageFactoryManager;
+  protected static WebDriver driver;
+  protected static DesiredCapabilities capabilities;
+  final protected int FIRST = 1;
+  final protected int TIMEOUT = 15;
 
- protected WebDriver driver;
-
-
-
-  @BeforeClass
-  public void TestSetUp() {
-    chromedriver().setup();
-    driver = new ChromeDriver();
+  @Parameters("browser")
+  @BeforeMethod
+  public void setUp(String browser) {
+    setUpDriver(browser);
     driver.manage().window().maximize();
     pageFactoryManager = new PageFactoryManager(driver);
-    homePage = pageFactoryManager.getHomePage();
-    alertAndPopupPage = pageFactoryManager.getAlertAndPopupPage();
+    initPages();
+  }
 
+  public void setUpDriver(String browser) {
+    if (browser.equalsIgnoreCase("chrome")) {
+      capabilities = DesiredCapabilities.chrome();
+    } else if (browser.equalsIgnoreCase("firefox")) {
+      capabilities = DesiredCapabilities.firefox();
+    } else if (browser.equalsIgnoreCase("MicrosoftEdge")) {
+      capabilities = DesiredCapabilities.edge();
+    }
+    try {
+      driver = new RemoteWebDriver(new URL("http://172.20.10.3:4444/"), capabilities);
```

Comment:
Please add the ability to run tests locally. For example, depending on the boolean localRun parameter you can implement remote or local Webdriver. This parameter can be taken from a general test suite runner or from a specific XML runner.